### ISSUE TITLE
Training model on yolo but -nan and nan

### ISSUE THREAD

**Issue by hamidfathi1998** [AlexeyAB/darknet#4848]
I want to train model on yolo but after one step give nan and -nan I have 300 images in different sizes (almost 600*600) and one class to detect Image. before I gave good results with 100 images (%75 accuracy of detection) but I want to give the best result.
http://tadbirserspota.ir/004.png

and I split 80-20 train and test data 

please help me!

tiny_yolo.cfg
```
[net]
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
max_batches = 120000
policy=steps
steps=-1,100,80000,100000
scales=.1,10,.1,.1

[convolutional]
batch_normalize=1
filters=30
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=30
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=30
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=30
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=30
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=30
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=30
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=30
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=30
activation=linear

[region]
anchors = 0.738768,0.874946,  2.42204,2.65704,  4.30971,7.04493,  10.246,4.59428,  12.6868,11.8741
bias_match=1
classes=1
coords=4
num=5
softmax=1
jitter=.2
rescore=1
small_object=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1

```

**Comment by stephanecharette** [comment]
Can you share a .zip file with the images and the markup?

**Comment by hamidfathi1998** [comment]
Thanks for your reply
Github/
hamidfathi1998/plateRes

**Comment by stephanecharette** [comment]
Your images are ok.  You are missing a few marks -- some images have multiple license plates, but you've only marked 1.  But most importantly, I think your .cfg file is incorrect.  You only filters=30 everywhere which I don't think is correct.  There are other possible issues, but I'm not super familiar with the internals of the .cfg files, so I don't know if those changes you made were intentional or not.  E.g., the steps, scales, and the custom anchors.  (I've never used custom anchors, so I cannot say if what you've done is correct.)

When I overwrite your .cfg with the "normal" yolov3-tiny, I can see training is happening fine.  After 2000 iterations, the mAP is around 95%.  This is what your network looks like:

![output](https://user-images.githubusercontent.com/5061352/74296386-79f63380-4cf8-11ea-93f1-af774a7d2b5f.png)

(If curious, the way I checked your images and so quickly built the network is with DarkMark and DarkHelp.)

**Comment by hamidfathi1998** [comment]
hi thanks for attention
can you send me .cfg file.
I do everything but did not work

‫‪Stéphane Charette‬‏ <‪notifications@github.com‬‏> در تاریخ چهارشنبه ۱۲
فوریهٔ ۲۰۲۰ ساعت ۵:۳۸ نوشت:‬

**Comment by stephanecharette** [comment]
> hi thanks for attention can you send me .cfg file.

I'll attach the .cfg file I used.  Especially when getting started, I recommend DarkMark to create the darknet project files.  Disclaimer:  I'm the author of DarkMark.

[iran_plates_yolov3-tiny.cfg.txt](https://github.com/AlexeyAB/darknet/files/4194746/iran_plates_yolov3-tiny.cfg.txt)

**Comment by hamidfathi1998** [comment]
thanks.
can I want you to send me weights?

**Comment by hamidfathi1998** [comment]
I use your .cfg file but give this error
http://tadbirserspota.ir/sasasasca.png

**Comment by AlexeyAB** [comment]
Currently you should compile Darknet with OpenCV to use mosaic=1 in cfg-file.

**Comment by hamidfathi1998** [comment]
thanks, that is work!
